### ISSUE TITLE
Registration form validation and required fields.

### ISSUE THREAD

**Issue by piet0024** [gctools-outilsgc/gccollab#114]
https://wet-boew.github.io/v4.0-ci/demos/formvalid/formvalid-en.html

The registration form labels would need some text that say the fields are required. 

The blue box mentioning using their proper name does not get read by the screen reader as they tab through the form. (this could be added by an aria-described by to provide helpful form hint)

The only error checking that happens when the form submits is the captcha.

@CoolColin

**Comment by markwooff** [comment]
Validation updated for missing fields. Also added aria tags for the blue description box and for required fields.
